quickjs.c

```diff
@@ -43821,6 +43835,24 @@ static JSValue js_regexp_escape(JSContext *ctx, JSValue this_val,
     return string_buffer_end(b);
 }
 
+static JSValue js_regexp_get_leftContext(JSContext *ctx, JSValue this_val) {
+    if (ctx->regexp_last_match_start >= 0) {
+        JS_FreeValue(ctx, ctx->regexp_left_ctx);
+        ctx->regexp_left_ctx = js_sub_string(ctx, JS_VALUE_GET_STRING(ctx->regexp_last_match_str), 0, ctx->regexp_last_match_start);
```

Comment:
We don't have a strict line length but try to keep it under 80 columns whenever you can, by doing something like:
```
JSString *p;
// ...
p = JS_VALUE_GET_STRING(ctx->regexp_last_match_str)
ctx->regexp_left_ctx = js_sub_string(ctx, p, 0, ctx->regexp_last_match_start);
```
Also, bad things will happen when `ctx->regexp_last_match_str` is _not_ a string. I realize that's guarded by the `ctx->regexp_last_match_start >= 0` check but it seems kind of fragile. Why not:
```c
if (JS_IsString(ctx->regexp_last_match_str)) {
    // ...
}
```